canary/webrtc-c/src/Peer.cpp

```diff
@@ -536,11 +536,16 @@ STATUS Peer::addTransceiver(RtcMediaStreamTrack& track)
         frameDataPtr += SIZEOF(UINT64);
         UINT32 receivedSize = getUnalignedInt32BigEndian((PINT32)(frameDataPtr));
 
-        pPeer->endToEndMetricsContext.frameLatency.push_back((DOUBLE)(GETTIME() - receivedTs) / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+        pPeer->endToEndMetricsContext.frameLatencyAvg =
+            EMA_ACCUMULATOR_GET_NEXT(pPeer->endToEndMetricsContext.frameLatencyAvg, GETTIME() - receivedTs);
```

Comment:
Just for my understanding. What is the hypothesis for EMA fixing the e2e metric emitting problem?

Reply:
Again, it's not the EMA or how it's calculated, but there's a data race in collecting and publishing these metrics. Therefore, I added this lock, https://github.com/aws-samples/amazon-kinesis-video-streams-demos/pull/122/files#diff-36c06b4e0a11b29790dda910ed5a0beb1cf833082548e8ebe1370d8be1c8e476R702.

EMA is an enhancement from what we had, and should be preferred as it's standard in our SDKs to calculate average in a stream.

Reply:
Ok. Makes sense.